airflow/providers/openlineage/extractors/bigquery.py

```diff
@@ -0,0 +1,108 @@
+    def get_operator_classnames(cls) -> list[str]:
+        return ["BigQueryOperator", "BigQueryExecuteQueryOperator"]
+
+    def extract(self) -> OperatorLineage | None:
+        return None
+
+    def extract_on_complete(self, task_instance) -> OperatorLineage | None:
+        self.log.debug("extract_on_complete(%s)", str(task_instance))
+
+        try:
+            bigquery_job_id = self._get_xcom_bigquery_job_id(task_instance)
+            if bigquery_job_id is None:
+                raise Exception("Xcom could not resolve BigQuery job id. Job may have failed.")
+        except Exception as e:
+            self.log.exception("Cannot retrieve job details from BigQuery.Client.", exc_info=True)
+            return OperatorLineage(
+                run_facets={
+                    "bigQuery_error": BigQueryErrorRunFacet(
+                        clientError=f"{e}: {traceback.format_exc()}",
+                    )
+                }
+            )
+
+        client = self._get_client()
+
+        stats = BigQueryDatasetsProvider(client=client).get_facets(bigquery_job_id)
+        inputs = stats.inputs
+        output = stats.output
+
+        for ds in inputs:
+            ds.input_facets = self._get_input_facets()
+
+        run_facets = stats.run_facets
+        job_facets = {"sql": SqlJobFacet(self.operator.sql)}
+
+        return OperatorLineage(
+            inputs=[ds.to_openlineage_dataset() for ds in inputs],
+            outputs=[output.to_openlineage_dataset()] if output else [],
+            run_facets=run_facets,
+            job_facets=job_facets,
+        )
+
+    def _get_client(self):
+        # lazy-load the bigquery Client due to its slow import
+        from google.cloud.bigquery import Client
+
+        # Get client using Airflow hook - this way we use the same credentials as Airflow
+        if hasattr(self.operator, "hook") and self.operator.hook:
+            hook = self.operator.hook
+            return hook.get_client(project_id=hook.project_id, location=hook.location)
+        with suppress(ImportError):
+            BigQueryHook = import_string("airflow.providers.google.cloud.operators.bigquery.BigQueryHook")
+
+        if BigQueryHook is not None:
+            hook = BigQueryHook(
+                gcp_conn_id=self.operator.gcp_conn_id,
+                use_legacy_sql=self.operator.use_legacy_sql,
+                delegate_to=self.operator.delegate_to,
+                location=self.operator.location,
+                impersonation_chain=self.operator.impersonation_chain,
+            )
+            return hook.get_client(project_id=hook.project_id, location=hook.location)
+        return Client()
+
+    def _get_xcom_bigquery_job_id(self, task_instance):
+        bigquery_job_id = task_instance.xcom_pull(task_ids=task_instance.task_id, key="job_id")
+
+        self.log.debug("bigquery_job_id: %s", str(bigquery_job_id))
+        return bigquery_job_id
+
+    def _get_input_facets(self):
+        return {}
```

Comment:
It seems like an important method, at least according to:
https://openlineage.io/blog/operators-and-extractors-technical-deep-dive/
But I didn't find any related documentation in-code.
It might be worth adding some.